Data/test.lua

```diff
@@ -0,0 +1,9646 @@
+        {
+          Category="Class",
+          Nature="AUTOMATIC",
+          Name="Barbarian",
+        },
+      },
+      WeaponBonusProficiencySelections={
+      },
+    },
+  },
+  Name="Ex-Barbarian",
+  Definitions={
+    {
+      Name="BarbarianLVL",
+      InitialValue=0,
+    },
+  },
+  Bonuses={
+    {
+      Category="COMBAT",
+      Variables={
+        "BASEAB",
+      },
+      Formula="classlevel(\"APPLIEDAS=NONEPIC\")",
+      Conditions={
+        function (character)
+          return (character.Variables["UseAlternateBABProgression"] == 0)
+        end,
+      },
+    },
+    {
+      Category="SAVE",
+      Variables={
+        "BASE.Fortitude",
+      },
+      Formula="classlevel(\"APPLIEDAS=NONEPIC\")/2+2",
+      Conditions={
+        function (character)
+          return (character.Variables["UseAlternateSaveProgression"] == 0)
+        end,
+      },
+    },
+    {
+      Category="SAVE",
+      Variables={
+        "BASE.Reflex",
+        "BASE.Will",
+      },
+      Formula="classlevel(\"APPLIEDAS=NONEPIC\")/3",
+      Conditions={
+        function (character)
+          return (character.Variables["UseAlternateSaveProgression"] == 0)
+        end,
+      },
+    },
+    {
+      Category="VAR",
+      Variables={
+        "ClassBABFull",
+      },
+      Formula="classlevel(\"APPLIEDAS=NONEPIC\")",
+      Conditions={
+        function (character)
+          return (character.Variables["UseFractionalBAB"] == 1)
+        end,
+      },
+    },
+    {
+      Category="VAR",
+      Variables={
+        "BarbarianLVL",
+      },
+      Formula="CL",
+    },
+    {
+      Category="VAR",
+      Variables={
+        "ClassSaveGood_Fortitude",
+      },
+      Formula="classlevel(\"APPLIEDAS=NONEPIC\")",
+      Conditions={
+        function (character)
+          return (character.Variables["UseFractionalSave"] == 1)
+        end,
+      },
+    },
+    {
+      Category="VAR",
+      Variables={
+        "ClassSavePoor_Reflex",
+      },
+      Formula="classlevel(\"APPLIEDAS=NONEPIC\")",
+      Conditions={
+        function (character)
+          return (character.Variables["UseFractionalSave"] == 1)
+        end,
+      },
+    },
+    {
+      Category="VAR",
+      Variables={
+        "ClassSavePoor_Will",
+      },
+      Formula="classlevel(\"APPLIEDAS=NONEPIC\")",
+      Conditions={
+        function (character)
+          return (character.Variables["UseFractionalSave"] == 1)
+        end,
+      },
+    },
+  },
+  WeaponBonusProficiencySelections={
+  },
+})
+DefineClass({
+  HitDie=8,
+  MaxLevel=20,
+  SkillPointsPerLevel=6,
+  Memorize=false,
+  Roles={
+    "Skill",
+  },
+  Types={
+    "Base",
+    "PC",
+    "SpontaneousArcane",
+    "Spontaneous",
+  },
+  Facts={
+    ClassType="PC",
+    Abb="Brd",
+    SpellType="Arcane",
+  },
+  Levels={
+    {
+      Level="1",
+      SpellsPerDay={
+        0, 1, },
+      SpellsKnown={
+        4, 2, },
+      WeaponBonusProficiencySelections={
```

Comment:
Are these empty things necessary? Seems like we could make this files quite a bit more compact by skipping empty stuff.  Or does somehow setting WeaponBonusProficiencySelections to an empty object do something different than not setting it at all (in which case I'm a little scared)

Reply:
This is here because I didn't condition writing the field on it being empty, I'll fix that.